compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -1085,7 +1085,10 @@ static void addLowerToLLVMGPUPasses(OpPassManager &modulePassManager,
   FunctionLikeNest funcPassManager(modulePassManager);
   funcPassManager.addPass(createFoldTensorExtractOpPass)
       .addPass(createLLVMGPUVectorLoweringPass)
-      .addPass(createExpandGPUOpsPass);
+      .addPass(createExpandGPUOpsPass)
+      .addPass(createResolveSwizzleHintsPass)
+      .addPass(createCanonicalizerPass)
```

Comment:
Drive-by comment: this might end up being a bit early? I think a bunch of the de-subviewing and such is in `lowerAndOptimizeAddressComputationPasses`, so perhaps this should be in there

Reply:
Makes sense, done. I hadn't used any subviews in the path I prototyped this with so that's perhaps why I hadn't hit this.